packages/typescript/src/plugins/FederatedTypesPlugin.ts

```diff
@@ -82,12 +89,22 @@ export class FederatedTypesPlugin {
     if (!disableDownloadingRemoteTypes) {
       compiler.hooks.beforeRun.tapAsync(PLUGIN_NAME, async (_, callback) => {
         this.logger.log('Preparing to download types from remotes on startup');
-        await this.importRemoteTypes(callback);
+        try {
+          await this.importRemoteTypes();
+          callback();
```

Comment:
It will return a promise due to the fact the function is marked with the async keyword, do you want all the requests rolled up?

Reply:
Wouldn't `.allSettled` with a `.catch` on the request work better? I feel that it would fix this issue https://github.com/module-federation/universe/issues/498

Reply:
Yeah definitely could work like that, maybe having an exclusion list or a way to flag a remote as non-typescript would be ideal. Let failures stop the process but exclude remotes that should be excluded

Reply:
@pavandv - any thoughts on the PR with the comments? As far as #498 goes, maybe we just add a new “excluded remotes” key to the config

Reply:
@RussellCanfield the function might return a promise but `importRemoteTypes` will be immediately resolved since there are no actual promises inside waiting for the network requests to resolve.

if the `if conditions` inside `importRemoteTypes` doesn't get truthy for a given remote, it will be marked as resolved for all the other remotes as well.

I think it would be best to wrap it in `promise.all`

Reply:
@pavandv - I might be confused here, the first if condition before the remoteUrls loop does return out but that is only for no remotes being configured which is intended.

The foreach here is a loop of promises, so each iteration is awaited, this code was the exact same as it was before? I pushed a small change to change the loop syntax, and I also ran a test (screenshot below) where the first if inside the loop fails early for a remote, and you can see the other remotes are still requested:

<img width="553" alt="image" src="https://user-images.githubusercontent.com/17344904/211857467-20ac8c90-e91c-4bc7-b5fd-5b6f0041be40.png">

Reply:
The problem I was mentioning is a little different but I think your recent change for the `forEach` loop has fixed the problem statement I was mentioning. 

This looks good to me.